### ISSUE TITLE
Where is the sparse regularization term?

### ISSUE THREAD

**Issue by xuxy09** [jimmy-ren/vcnn_double-bladed#1]
There should be a sparse regularization term in the loss function according to your ICML2015 paper. But it seems there is not such a term either in the cost function in forward pipeline or in the derivative of the cost function in backprop pipeline. Have you leave out that term in the released code?

**Comment by jimmy-ren** [comment]
Thanks for the question. The way I implemented it is to add |X| (absolute value of X, where X is the network output) in the loss function. The implementation of its backprop is straightforward if you consider it as |X| as sqrt(X^2). I'll probably add it back once I have time. If you have this term, you will have a faster convergence.

**Comment by xuxy09** [comment]
Thanks for your reply. I guess I can add the backprop term by modifying the file 'out_backprop.m' directly, right?
Actually, the reason why I have this question is that my result of L0 smoothing filter (lamda=0.08) is not very good. So I doubt whether this missing term matters. But according to your reply, this sparse regularization term seems only to affect the convergence speed rather than the filter effect. I use the BSDS500 dataset, and my validation cost after 200 epoc is about 6, is this cost too large so that I need more iterations? Or I should use a larger dataset like Flickr instead of BSDS500?

**Comment by jimmy-ren** [comment]
1. Yes.
2. I haven't tried lamda=0.08 for L0 smoothing filter. You may use this sparsity term and have a try. I don't think the problem is with the dataset. If it still can't approximate this stronger filter, I guess a stronger (deeper) network architecture need to be used.

**Comment by xuxy09** [comment]
Thanks! The result turns to be better now after training again using the same code and data. I guess my first time training may have a bad weight initialization for the weight is initialized randomly.
